packages/zipkin/src/instrumentation/httpServer.js

```diff
@@ -51,14 +51,14 @@ class HttpServerInstrumentation {
         });
       });
     } else {
-      if (readHeader(Header.Flags) !== None) {
+      if (readHeader(Header.Flags) !== None || readHeader(Header.Sampled) !== None) {
         const currentId = this.tracer.id;
         const idWithFlags = new TraceId({
-          traceId: currentId.traceId,
-          parentId: currentId.parentId,
-          spanId: currentId.spanId,
-          sampled: currentId.sampled,
-          flags: readHeader(Header.Flags)
+          traceId: new Some(currentId.traceId),
+          parentId: new Some(currentId.parentId),
```

Comment:
is some right here? Can't parentId be absent?

Reply:
looks like the current behavior returns the span ID when the parent is absent, which is kindof surprising.

your code will work, but I think it is walking around a general flaw here. parentId return an Option even if it is consistently not an option today (ex in typescript https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin/index.d.ts#L61)

@eirslett need a second opinion on this which amounts to a separate issue I suppose if the behavior is not desirable. Note: I've had to do some defense in the UI as sometimes people have written json with spanId == parentId accidentally.